cpu/stm32/periph/gpio_all.c

```diff
@@ -246,6 +270,31 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     else {
         NVIC_EnableIRQ(EXTI4_15_IRQn);
     }
+#elif defined(CPU_FAM_STM32MP1)
+    if (pin_num < 5) {
+        NVIC_EnableIRQ(EXTI0_IRQn + pin_num);
+    }
+    else if (pin_num < 6) {
+        NVIC_EnableIRQ(EXTI5_IRQn);
+    }
+    else if (pin_num < 10) {
+        NVIC_EnableIRQ(EXTI6_IRQn + pin_num - 6);
+    }
+    else if (pin_num < 11) {
+        NVIC_EnableIRQ(EXTI10_IRQn);
+    }
+    else if (pin_num < 12) {
+        NVIC_EnableIRQ(EXTI11_IRQn);
+    }
+    else if (pin_num < 14) {
+        NVIC_EnableIRQ(EXTI12_IRQn + pin_num - 12);
+    }
+    else if (pin_num < 15) {
+        NVIC_EnableIRQ(EXTI14_IRQn);
+    }
+    else {
+        NVIC_EnableIRQ(EXTI15_IRQn);
+    }
```

Comment:
This would lend itself to a switch-case block.

Reply:
ack

Reply:
Tested values are not constant but ranges.
Thus it would lead to something that seems to me less readable like:
```
switch (pin_num) {
    case 0:
    case 1:
    case 2:
    case 3:
    case 4:
        NVIC_EnableIRQ(EXTI0_IRQn + pin_num);
        break;
...
```